File: memory-common/src/main/java/com/memory/response/ServerResponse.java
package com.memory.response;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
@AllArgsConstructor
public class ServerResponse<T> {

    private Integer statusCode;
    private String message;
    private T data;

    public static final ServerResponse<String> OK = new ServerResponse<>(200, "OK", null);

    public static <T> ServerResponse<T> success(T data) {
        return new ServerResponse<>(200, "OK", data);
    }

    public static ServerResponse<Object> error(Integer statusCode, String message) {
        return new ServerResponse<>(statusCode, message, null);
    }

}

File: memory-common/src/main/kotlin/com/memory/response/ServerResponse.kt
package com.memory.response

data class ServerResponse<T>(
    val statusCode: Int,
    val message: String,
    val data: T?
) {
    companion object {
        @JvmField
        val OK = ServerResponse(200, "OK", null)

        @JvmStatic
        fun <T> success(data: T): ServerResponse<T> {
            return ServerResponse(200, "OK", data)
        }

        @JvmStatic
        fun error(statusCode: Int, message: String): ServerResponse<Any?> {
            return ServerResponse(statusCode, message, null)
        }
    }
}
